### ISSUE TITLE
VolumeRaycaster show index is out of range

### ISSUE THREAD

**Issue by woxiarong** [mlavik1/UnityVolumeRendering#273]
when "dataset.GetData(index.x, index.y, index.z) " to get the value, the console show "data[]"'s index is out of range.  Is that shuld check the "dataset.data.Length" and "x + y * dimX + z * (dimX * dimY)]"'s value before get the dataset's value ?

**Comment by mlavik1** [comment]
Good point. Maybe I should add a check, and add another `GetDataUnsafe` for the cases where we don't want to spend time on range checking..
Thanks for the suggestion! I'll keep this task open until that's done.

**Comment by woxiarong** [comment]
你的邮件我收到了，会尽快查看的^_^

**Comment by mlavik1** [comment]
好的 👌
可是我没有发送邮件 😅
